app/common/elasticsearchclient.py

```diff
@@ -668,33 +669,78 @@ def get_gene_info(self, gene_ids, **kwargs):
             q.size = len(gene_ids)
             q['from'] = params.start_from
 
+            if params.facets == 'true':
+                # q.aggregations.go_terms.terms.field = "go.value.term"
+                q.aggregations.significant_go_terms.significant_terms.field = "go.id.keyword"
+                # q.aggregations.significant_go_terms.significant_terms.field = "go.value.term"
+                # q.aggregations.significant_go_terms.significant_terms.include = "p:*|c:*|f:.*"
+                # q.aggregations.significant_go_terms.significant_terms.size = "15"
+                q.aggregations.significant_go_terms.aggregations.top_hits_goterms.top_hits.size= 1
+                q.aggregations.significant_go_terms.aggregations.top_hits_goterms.top_hits._source = ['go']
+            # if params.facets:
+            #     q.aggregations.go_terms.terms.field = "go.value.term"
+            #     q.aggregations.go_terms.terms.include = "p:*|c:*|f:.*"
+            #     q.aggregations.go_terms.terms.size = 25
+
+            if params.go_term:
+                q.post_filter.term['go.id.keyword'] = params.go_term
+
+            if params.fields:
+                q._source = params.fields
+
             res = self._cached_search(index=self._index_genename,
                                       doc_type=self._docname_genename,
                                       body=q.to_dict())
 
-            if res['hits']['total']:
-                return SimpleResult(res, params)
+            if 'aggregations' in res:
+                res['aggregations']['significant_go_terms']['buckets']  = self._process_go_info(res['aggregations']['significant_go_terms']['buckets'])
+
+            return PaginatedResult(res,params)
+
+    def _process_go_info(self,bucket_list):
+        go_term_buckets = list()
+        for bucket in bucket_list:
+            bucket_dict = dict()
+            bucket_dict['key'] = bucket['key']
+            bucket_dict['doc_count'] = bucket['doc_count']
+
+            term = [go['value']['term'] for go in bucket['top_hits_goterms']['hits']['hits'][0]['_source']['go'] if go['id'] == bucket_dict['key']]
+            label = re.findall(r'(\w):([\w+\s]+[\w+])', term[0])
```

Comment:
what are you filtering here? if it's only 3 cases (ie. the f, p, c labels) you don't need the regex

Reply:
following @emepyc comment: we could 
```python
if label == `f` then:
    include it
else:
    ignore it
```